itest/rust/src/packed_array_test.rs

```diff
@@ -200,3 +202,41 @@ fn packed_array_reverse() {
     array.reverse();
     assert_eq!(array.to_vec(), vec![2, 1]);
 }
+
+#[itest]
+fn packed_array_as_slice() {
+    let vec = vec![1, 2];
+    let array = PackedByteArray::from(vec.as_slice());
+    assert_eq!(array.as_slice(), vec.as_slice());
+
+    assert_eq!(
+        PackedByteArray::new().as_slice(),
+        &[],
+        "array is empty, but still a valid slice"
+    );
+}
+
+#[itest]
+fn packed_array_is_mut_unique() {
+    let array1 = PackedByteArray::from(&[1, 2]);
+    let mut array2 = array1.clone();
+
+    assert_eq!(
+        array1.as_slice().as_ptr(),
+        array2.as_slice().as_ptr(),
+        "arrays should share the same buffer"
+    );
+
+    // array2 should become a copy of array1 since rc > 1.
```

Comment:
The "RC > 1" is somewhat irrelevant, now that Godot 4 is released.

I would rather mention that `as_mut_slice` triggers a copy-on-write.